python/vowpalwabbit/DFtoVW.py

```diff
@@ -789,55 +926,48 @@ def check_missing_columns_df(self):
         else:
             missing_cols["tag"] = tag_columns - df_colnames
 
-        all_features = [
-            feature
-            for namespace in self.namespaces
-            for feature in namespace.features
-        ]
+        all_features = [feature for namespace in self.namespaces for feature in namespace.features]
         missing_features_cols = set()
         for feature in all_features:
             missing_features_cols.update(feature.columns - df_colnames)
 
-        missing_cols["Feature"] = sorted(list(missing_features_cols))
+        missing_cols["Feature"] = missing_features_cols
 
         missing_cols = {
-            key: value
-            for (key, value) in missing_cols.items()
-            if len(value) > 0
+            key: sorted(list(value)) for (key, value) in missing_cols.items() if len(value) > 0
         }
-        self.generate_missing_col_error(missing_cols)
+        if missing_cols:
+            self.raise_missing_col_error(missing_cols)
 
-    def generate_missing_col_error(self, absent_cols_dict):
-        """Generate error if some columns are missing
+    def raise_missing_col_error(self, missing_cols_dict):
```

Comment:
use **raise** instead of **generate**. Also this function is now called only if there are missing columns (see method `check_missing_columns_df` ).